spec/unit/cli/cli_spec.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+
+require 'spec_helper'
+require 'test_utils'
+
+
+describe Hocon::CLI do
+  ####################
+  # Argument Parsing
+  ####################
+  context 'argument parsing' do
+    it 'should find all the flags and arguments' do
+      args = %w(-i foo -o bar set some.path some_value --json)
+      expected_options = {
+        in_file: 'foo',
+        out_file: 'bar',
+        subcommand: 'set',
+        path: 'some.path',
+        new_value: 'some_value',
+        json: true
+      }
+      expect(Hocon::CLI.parse_args(args)).to eq(expected_options)
+    end
+  end
+
+  context 'subcommands' do
+    hocon_text =
+'foo.bar {
+  baz = 42
+  array = [1, 2, 3]
+  hash: {key: value}
+}'
+
+    context 'do_get()' do
+      it 'should get simple values' do
+        options = {path: 'foo.bar.baz'}
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq('42')
+      end
+
+      it 'should work with arrays' do
+        options = {path: 'foo.bar.array'}
+        expected = "[\n    1,\n    2,\n    3\n]"
+        expect(Hocon::CLI.do_get(options, hocon_text)).to eq(expected)
+      end
+
+      it 'should work with hashes' do
+        options = {path: 'foo.bar.hash'}
+        expected = "key=value\n"
```

Comment:
ok, i'm good w/o outer braces.  I think it would be worth a couple of hours of poking to see if there was some way to add an optional "RenderOptions" class, that would allow you to toggle certain things (like `=` vs. `:`, and omitting the `:` before a map).  If we could figure out an easy way to add that, I think it might be nice to have the tool default to passing in a RenderOptions object that would make the output more like our current files.

We could discuss whether or not that should be the default, I guess, but in any case it seems like investigating the feasibility of a RenderOptions thing would be worth it.  If it turns out to be too hairy we can probably punt on it for now.